e2e/test_operator_upgrades/operator_upgrades_test.go

```diff
@@ -159,8 +160,9 @@ var _ = Describe("Operator Upgrades", Label("e2e", "pr"), func() {
 			Expect(err).NotTo(HaveOccurred())
 
 			// During an upgrade we expect that the transaction system processes are replaced, so we expect to have seen
-			// 2 times the process counts for transaction system processes.
-			expectedProcessCounts := (processCounts.Total() - processCounts.Storage) * 2
+			// 2 times the process counts for transaction system processes. Add a small buffer of 5 to allow automatic
+			// replacements during an upgrade.
+			expectedProcessCounts := (processCounts.Total()-processCounts.Storage)*2 + 5
```

Comment:
Same buffer count in both HA and non-HA cases?

Reply:
We can reduce the buffer for the non-HA case. The buffer is still small enough for the non-HA case to catch issues where we are doing 2 replacements of the whole transaction system Pods.